main/zoe/api/zoe-contract-facet.md

```diff
@@ -415,39 +424,29 @@ a valid keyword, or is not unique.
 ```js
 zcf.assertUniqueKeyword(keyword);
 ```
-## `zcf.reallocate(seatStagings)`
-- `seatStagings` `{SeatStaging[]}` (at least two)
+## `zcf.reallocate(seats)`
+- `seats` `{ZCFSeats[]}` (at least two)
 - Returns: `{void}`
 
-The contract reallocates over `seatStagings`, which are
-associations of `seats` with new `allocations` to be used in reallocation.
-There must be at least two `seatStagings` in the array argument.
+The contract reallocates over its `seats` arguments, which are
+associations of `seats` with staged allocations to be used in reallocation.
+There must be at least two `ZCFseats` in the array argument. Every `ZCFSeat`
+with a staged allocation must be included in the argument array or an error
+is thrown. If any seat in the argument array does not have a staged allocation,
+an error is thrown. 
+
+On commit, the staged allocations become the seats' current allocations and 
+the staged allocations are deleted.
+
+Note: `reallocate()` is an *atomic operation*. To enforce offer safety, 
+it will never abort part way through. It will completely succeed or it will 
+fail before any seats have their current allocation changed
 
 The reallocation only succeeds if it:
 1. Conserves rights (the specified `amounts` have the same total value as the
   current total amount)
 2. Is 'offer-safe' for all parties
   involved. Offer safety is checked at the staging step.
 
-The reallocation is partial, only applying to `seats` associated
-with the `seatStagings`. By induction, if rights conservation and
-offer safety hold before, they hold after a safe reallocation.
```

Comment:
We need to say something new about offer safety.  The existing versions verify offer safety as each staging is created, but the new approach is unable to do that, so offer safety is checked on `reallocate`, and can cause the transaction to fail.

Reply:
I don't think we need to say anything new about offer safety. We just need to remove anything that says "offer safety is checked at staging". Checking offer safety at the reallocation step is the model we had originally, and I don't think any contracts will have problems moving back to that model (nor will they notice).